### PR TITLE
Javascript/javascript1/week4

### PR DESCRIPTION
**Pull request by Sarulathaanbu**

### CODE REVIEW

File: javascript/javascript1/week4/voiceassistant.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  ////////////Timer Setup//////////////////////
+  function timer() {
+    console.log(`Timer done`);
+  }
+  setTimeout(timer, 6000);
+  
+  console.log("Timer set for 1 minute");
+  console.log(getReply("Hello my name is Benjamin")); // "Nice to meet you benjamin"
+  console.log(getReply("What is my name?")); // "Your name is Benjamin"
+  console.log(toDoList("Add fishing to my todo"));
```
Comment (FrederikNygart) [inline javascript/javascript1/week4/voiceassistant.js:69]: To correctly solve the homework assignment you to call only `getReply("input")`

`console.log(toDoList("Add fishing to my todo"));` is therefore not allowed, and neither `console.log(calculate(29, 20, "-"));`

**Review by FrederikNygart — COMMENTED** [review]
Hi Saru, please let me know on slack if you need my help.

File: javascript/javascript1/week4/voiceassistant.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    if (operator == "+") {
+      return firstNumber + secondNumber;
+    }
+    if (operator == "-") {
+      return firstNumber - secondNumber;
+    }
+
+    ////////////Timer Setup function calling//////////////////////
+    if (command.includes("Set Timer")) {
```
Comment (FrederikNygart) [inline javascript/javascript1/week4/voiceassistant.js:61]: this should be flexible enough to understand different amounts and units

File: javascript/javascript1/week4/voiceassistant.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    if (command.includes("What is my name?")) {
+      return `My name is ${myName}`;
+    }
+
+    /////////////////Today's Date///////////////
+    if (command.includes("What day is it today?")) {
+      return (new Date().toDateString());
+    }
+
+    ////////////////// ToDoList Exercise////////////////
```
Comment (FrederikNygart) [inline javascript/javascript1/week4/voiceassistant.js:25]: instead of comments consider extracting functionality into smaller functions with fitting names

**Review by FrederikNygart — CHANGES_REQUESTED** [review]

File: javascript/javascript1/week4/voiceassistant.js (hunk 2 above)
Comment (Sarulathaanbu) [reply javascript/javascript1/week4/voiceassistant.js]: Changes done

File: javascript/javascript1/week4/voiceassistant.js (hunk 3 above)
Comment (Sarulathaanbu) [reply javascript/javascript1/week4/voiceassistant.js]: Created smaller functions

File: javascript/javascript1/week4/voiceassistant.js (hunk 1 above)
Comment (Sarulathaanbu) [reply javascript/javascript1/week4/voiceassistant.js]: Called in getReply("input")

**Review by FrederikNygart — APPROVED** [review]
Really good job! One for the road: Variable and function naming is important. Remember to consider the name that best and most consisely describes *what* data a function gets/sets/stores/changes/displays. It is not easy, but practice makes perfect.

**Pull request merged**